libraries/classes/ConfigStorage/Relation.php

```diff
@@ -1554,64 +1551,47 @@ public function fixPmaTables(string $db, bool $create = true): void
+            $createQueries = $this->getCreateTableSqlQueries($tableNameReplacements);
+            if (! $this->dbi->selectDb($db, Connection::TYPE_CONTROL)) {
+                $GLOBALS['message'] = $this->dbi->getError(Connection::TYPE_CONTROL);
 
-            // Default table name, nothing to do
-            if ($GLOBALS['cfg']['Server'][$feature] === $table) {
-                continue;
+                return;
             }
-
-            // Set the replacement to transform the default table name into a custom name
-            $tableNameReplacements[$table] = $GLOBALS['cfg']['Server'][$feature];
         }
 
-        $createQueries = null;
         $foundOne = false;
         foreach ($tablesToFeatures as $table => $feature) {
             // Check if the table already exists
             // use the possible replaced name first and fallback on the table name
             // if no replacement exists
             if (! in_array($tableNameReplacements[$table] ?? $table, $existingTables)) {
-                if ($create) {
-                    if ($createQueries == null) { // first create
-                        $createQueries = $this->getDefaultPmaTableNames($tableNameReplacements);
-                        if (! $this->dbi->selectDb($db, Connection::TYPE_CONTROL)) {
-                            $GLOBALS['message'] = $this->dbi->getError(Connection::TYPE_CONTROL);
-
-                            return;
-                        }
-                    }
-
-                    $this->dbi->tryQuery($createQueries[$table], Connection::TYPE_CONTROL);
+                if (! $create) {
+                    continue;
+                }
 
-                    $error = $this->dbi->getError(Connection::TYPE_CONTROL);
-                    if ($error) {
-                        $GLOBALS['message'] = $error;
+                $this->dbi->tryQuery($createQueries[$table], Connection::TYPE_CONTROL);
 
-                        return;
-                    }
+                $error = $this->dbi->getError(Connection::TYPE_CONTROL);
+                if ($error) {
+                    $GLOBALS['message'] = $error;
 
-                    $foundOne = true;
-                    if (empty($GLOBALS['cfg']['Server'][$feature])) {
-                        // Do not override a user defined value, only fill if empty
-                        $GLOBALS['cfg']['Server'][$feature] = $table;
-                    }
-                }
-            } else {
-                $foundOne = true;
-                if (empty($GLOBALS['cfg']['Server'][$feature])) {
-                    // Do not override a user defined value, only fill if empty
-                    $GLOBALS['cfg']['Server'][$feature] = $table;
+                    return;
                 }
             }
+
+            $foundOne = true;
+
+            // Do not override a user defined value, only fill if empty
+            if (isset($GLOBALS['cfg']['Server'][$feature]) && $GLOBALS['cfg']['Server'][$feature] !== '') {
```

Comment:
Hey, just a thought, won't this ignore if feature is set to `false` (disabled)?

[https://docs.phpmyadmin.net/en/latest/config.html#cfg_Servers_tracking](url)

I think one more additional check is needed

```suggestion
            if (isset($GLOBALS['cfg']['Server'][$feature]) && $GLOBALS['cfg']['Server'][$feature] !== '' && $GLOBALS['cfg']['Server'][$feature] !== false) {
```

What do you think?